marlowe/src/Language/Marlowe/Util.hs

```diff
@@ -65,7 +65,9 @@ foldMapContract fcont fcase fobs fvalue contract =
         Assert obs cont      -> fobs' obs <> go cont
   where
     go = foldMapContract fcont fcase fobs fvalue
-    fcase' cs@(Case _ cont) = fcase cs <> go cont
+    fcase' cs = fcase cs <> case cs of
+        Case _ cont        -> go cont
+        MerkleizedCase _ _ -> mempty
```

Comment:
Should the semantics of `foldMapContract` make a special provision for handling merkleization? For instance, it could include an argument `fmerk :: SerialisedContinuation -> m` so that `MerkleizedCase _ _ -> mempty` would become `MerkleizedCase _ cont -> fmerk cont`. One could always recover the behavior here by setting `fmerk = const mempty`, but more complex code could use `fmerk` to lookup and deserialise the contract corresponding to the hash.

Reply:
Yes, you are right, it can be quite problematic, I think I addressed it now. I am also a bit nervous about `contractLifespanUpperBound`, but we can think what to do with it later